File: Go/1004. Max Consecutive Ones III.go
package main

//goland:noinspection GoUnusedFunction
func longestOnes(nums []int, k int) int {
	left := 0
	right := 0

	for ; right < len(nums); right++ {
		if nums[right] == 0 {
			k--
		}

		if k < 0 {
			if nums[left] == 0 {
				k++
			}
			left++
		}
	}

	return right - left
}

File: PHP/1004. Max Consecutive Ones III.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $nums
     */
    public function longestOnes(array $nums, int $k): int
    {
        $left = 0;
        $right = 0;

        for (; $right < count($nums); $right++) {
            if ($nums[$right] == 0) {
                $k--;
            }

            if ($k < 0 && $nums[$left++] == 0) {
                $k++;
            }
        }

        return $right - $left;
    }
}
